src/App.tsx

```diff
@@ -14,11 +14,128 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+enum SortBy {
+  none,
+  name,
+  length,
+}
+
+type State = {
+  goods: string[];
+  isVisible: boolean;
+  isReversed: boolean;
+  sortBy: SortBy;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    isReversed: false,
+    sortBy: SortBy.none,
+  };
+
+  start = () => {
+    this.setState({ isVisible: true });
+  };
+
+  reverse = () => {
+    this.setState({ isReversed: true });
+  };
```

Comment:
You could have take the prevState and make it !prevState so that reverse() works multiple times...

this.setState((prevState) => (
      { isReversed: !prevState.isReversed }
));